pkg/controllers/clusterresourceplacement/controller.go

```diff
@@ -787,10 +774,28 @@ func buildResourcePlacementStatus(crp *fleetv1beta1.ClusterResourcePlacement, la
 		if c.ClusterScore != nil {
 			scheduledCondition.Message = fmt.Sprintf(resourcePlacementConditionScheduleSucceededWithScoreMessageFormat, c.ClusterName, c.ClusterScore, c.Reason)
 		}
-
-		// TODO populate the work status
 		meta.SetStatusCondition(&rp.Conditions, scheduledCondition)
-		res = append(res, rp)
+		if err := r.setWorkStatusForResourcePlacementStatus(ctx, crp, latestResourceSnapshot, &rp); err != nil {
+			return err
+		}
+		placementStatuses = append(placementStatuses, rp)
+
+		syncCondition := meta.FindStatusCondition(rp.Conditions, string(fleetv1beta1.ResourceWorkSynchronizedConditionType))
```

Comment:
iirc, this is reading from the RP status which was set in function call in line 778? If so, why not return it from the function call directly?